filter.h

```diff
@@ -0,0 +1,72 @@
+// Filter Module
```

Comment:
Add documentation for the whole file and make sure to use @defgroup and @{ so it gets treated as a module. Also consider adding an example showing how it could be used with the ADC module (use @code and @endcode to make it look nice)

Reply:
also, doxygen blocks start with /** so make sure anything you want included in the documentation starts with /**